protocol/src/main/scala/io/mediachain/protocol/CborSerialization.scala

```diff
@@ -601,6 +606,45 @@ object CborSerialization {
         )
   }
 
+  object JournalBlockArchiveDeserializer extends CborDeserializer[JournalBlockArchive]
+  {
+    def fromCMap(cMap: CMap): Xor[DeserializationError, JournalBlockArchive] =
+      for {
+        _ <- assertRequiredTypeName(cMap, MediachainTypes.JournalBlockArchive)
+        ref <- getRequiredReference(cMap, "ref")
+        xblock <- getRequired[CMap](cMap, "block")
+        block <- JournalBlockDeserializer.fromCMap(xblock)
+        xdata <- getRequired[CArray](cMap, "data")
+        data <- dataFromCbor(xdata)
+      } yield
+        JournalBlockArchive(
+          ref = ref,
+          block = block,
+          data = data
+        )
+    
+    def dataFromCbor(records: CArray) = {
+      def loop(rest: List[CValue], map: Map[Reference, Array[Byte]])
+      : Xor[DeserializationError, Map[Reference, Array[Byte]]] = {
+        rest match {
+          case CArray(List(refCbor: CMap, data: CBytes)) :: rest =>
+            ReferenceDeserializer.fromCMap(refCbor) match {
+              case Xor.Right(ref) =>
+                loop(rest, map + (ref -> data.bytes))
```

Comment:
I'm kind of surprised you didn't `reverse` the list and build this through normal returns from the recursion